src/core/hard_forks.zig

```diff
@@ -0,0 +1,56 @@
+const std = @import("std");
+const ArrayList = std.ArrayList;
+const allocator = std.mem.Allocator;
+const Slot = @import("slot.zig").Slot;
+
+pub const HardFork = struct { Slot, usize };
+pub const HardForks = struct {
+    hard_forks: ArrayList(HardFork),
+
+    const Self = @This();
+    pub fn default(alloc: allocator) Self {
+        return .{ .hard_forks = ArrayList(HardFork).init(alloc) };
+    }
+    pub fn register(self: *Self, new_slot: Slot) !void {
+        const index = for (self.hard_forks.items, 0..) |hf, i| {
+            if (hf[0].value == new_slot.value) break i;
+        } else null;
+
+        if (index != null) {
+            try self.hard_forks.append(.{ new_slot, self.hard_forks.items[index.?][1] +| 1 });
+        } else {
+            try self.hard_forks.append(.{ new_slot, 1 });
+        }
+        std.mem.sort(HardFork, self.hard_forks.items, {}, struct {
+            pub fn lessThan(_: void, a: HardFork, b: HardFork) bool {
+                return a[0].value < b[0].value;
+            }
+        }.lessThan);
+    }
+    pub fn deinit(self: *Self) void {
+        self.hard_forks.deinit();
+    }
+    pub fn get_forks(self: *Self) []HardFork {
+        return self.hard_forks.items;
+    }
+};
+
+test "hard_forks" {
```

Comment:
can we change this to "core.hard_forks: test hardforks" to follow the test naming standards

Reply:
Yea will do and would be good to add these to contributing.md